akka-remote/src/main/scala/akka/remote/serialization/ArteryMessageSerializer.scala

```diff
@@ -93,42 +93,45 @@ private[akka] final class ArteryMessageSerializer(val system: ExtendedActorSyste
   def deserializeQuarantined(quarantined: ArteryControlFormats.Quarantined): Quarantined =
     Quarantined(deserializeUniqueAddress(quarantined.getFrom), deserializeUniqueAddress(quarantined.getTo))
 
-  def serializeActorRef(ref: ActorRef): String = Serialization.serializedActorPath(ref)
-  def deserializeActorRef(str: String): ActorRef = system.provider.resolveActorRef(str)
+  def serializeActorRef(ref: ActorRef): String =
+    if ((ref eq ActorRef.noSender) || (ref eq system.deadLetters)) ""
```

Comment:
well, we can see it as if a real `deadLetters` ref sneaks in here we serialize/deserialize it correctly. Java serialization of `DeadLetterActorRef` is also special.

btw, I should perhaps change that `eq` check with `isInstanceOf[DeadLetterActorRef`?